kola/tests/ignition/resource.go

```diff
@@ -147,6 +147,40 @@ func init() {
 	      }`),
 		Distros: []string{"cl", "rhcos"},
 	})
+	// TODO: once Ignition supports this on all channels/distros
+	//       this test should be rolled into coreos.ignition.v2_1.resources.remote
+	// Test specifically for versioned s3 objects
+	register.Register(&register.Test{
+		Name:        "coreos.ignition.v2_1.resource.s3.versioned",
+		Run:         resourceS3Versioned,
+		ClusterSize: 1,
+		UserData: conf.Ignition(`{
+		  "ignition": {
+		      "version": "2.1.0"
+		  },
+		  "storage": {
+		      "files": [
+			  {
+			      "filesystem": "root",
+			      "path": "/var/resource/original",
+			      "contents": {
+				  "source": "http://s3-us-west-2.amazonaws.com/kola-fixtures/resources/versioned?versionId=null"
+			      },
+			      "mode": 420
+			  },
+			  {
+			      "filesystem": "root",
+			      "path": "/var/resource/latest",
+			      "contents": {
+				  "source": "http://s3-us-west-2.amazonaws.com/kola-fixtures/resources/versioned?versionId=RDWqxfnlcJOSDf1.5jy6ZP.oK9Bt7_Id"
```

Comment:
Do we have content in this resource that changed across versions and that we can check once the node is provisioned?

Reply:
Yes, in the original version (where `versionId` is `null`), the contents of the file are `original`, while in this version the contents are `updated`, see lines 245-248.